src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCaConfigurationTest.java

```diff
@@ -215,31 +210,29 @@ void Given_CustomCAConfiguration_WHEN_issuingAClientCertificate_THEN_itsSignedBy
         doReturn(chain).when(certificateStoreSpy).loadCaCertificateChain(privateKeyUri, certificateUri);
 
         AtomicReference<CertificateUpdateEvent> eventRef = new AtomicReference<>();
-        Pair<CompletableFuture<Void>, Consumer<CertificateUpdateEvent>> asyncCall =
-                TestUtils.asyncAssertOnConsumer(eventRef::set, 2);
         GetCertificateRequest request = buildCertificateUpdateRequest(
-                GetCertificateRequestOptions.CertificateType.CLIENT, asyncCall.getRight());
+                GetCertificateRequestOptions.CertificateType.CLIENT, eventRef::set);
 
         givenNucleusRunningWithConfig("config.yaml");
         subscribeToCertificateUpdates(request);
         givenCDAWithCustomCertificateAuthority(privateKeyUri, certificateUri);
 
-        asyncCall.getLeft().get(5, TimeUnit.SECONDS);
-        X509Certificate issuedClientCertificate = eventRef.get().getCertificate();
-        assertTrue(
-            CertificateTestHelpers.wasCertificateIssuedBy(intermediateCA, issuedClientCertificate),
-            String.format(
-                "Certificate not was not issued by intermediate CA %s", CertificateHelper.toPem(issuedClientCertificate)
-        ));
+        TestHelpers.eventuallyTrue(() -> {
```

Comment:
side thought---instead of busy-waiting, could introduce a new "custom ca configured" event emitted by `ConfigureCustomCertificateAuthority` and expose a subscription to the event via `ClientDevicesAuthServiceApi` or something.  would be a decent amount of work just for the sake of tests though 😅

Reply:
Yeah I don't like adding stuff that we are only using for tests if we can avoid it. If we eventually have a usecase to expose that then I agree